### PR TITLE
Allow to use HA Scripts as Button Entity

### PR DESCRIPTION
**Pull request by splattner**

### CODE REVIEW

File: src/client/service/mod.rs [hunk 1]
```diff
@@ -22,6 +22,7 @@ mod cover;
 mod light;
 mod media_player;
 mod switch;
+mod button;
```
Comment (zehnm) [inline src/client/service/mod.rs:25]: Please fix formatting. See `cargo fmt --all -- --check` output (or GitHub action).

**Review by zehnm — CHANGES_REQUESTED** [review]
Thank you for your contribution! The script support is working fine 👍 
Please fix the cosmetic formatting issues reported by Clippy, then I will merge this PR.

The only thing we could discuss is to use a switch instead of a button for HA scripts. This would allow to:
- stop a running script (at least according to the HA docs. I don't know if the script will be killed, or if it needs specific support to be stoppable).
- see when a (long running) script is active on the remote.

For me personally it doesn't really matter right now. I still have to migrate everything to HA and I'm not using scripts yet (just a matter of time...).

File: src/client/service/button.rs [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+use uc_api::ButtonCommand;
+
+pub(crate) fn handle_button(msg: &EntityCommand) -> Result<(String, Option<Value>), ServiceError> {
+    let cmd: ButtonCommand = cmd_from_str(&msg.cmd_id)?;
+
+    let entity: Vec<&str> = msg.entity_id.split('.').collect();
+
+    let service_call: &str = match entity[0] {
+            "script" => entity[1],
+            &_ => "press"
```
[the remark this answers is not in the document]
Comment (splattner) [reply src/client/service/button.rs]: sure, I just did

File: src/client/service/mod.rs (hunk 1 above)
Comment (splattner) [reply src/client/service/mod.rs]: also fixed

**Comment by splattner** [comment]
> Thank you for your contribution! The script support is working fine 👍 Please fix the cosmetic formatting issues reported by `cargo fmt`, then I will merge this PR. You can ignore the reported issue by Clippy, I've fixed that on the main branch.
> 
> The only thing we could discuss is to use a switch instead of a button for HA scripts. This would allow to:
> 
> * stop a running script (at least according to the HA docs. I don't know if the script will be killed, or if it needs specific support to be stoppable).
> * see when a (long running) script is active on the remote.
> 
> For me personally it doesn't really matter right now. I still have to migrate everything to HA and I'm not using scripts yet (just a matter of time...).

thanks for reviewing it. Fmt errors are fixed, still new to Rust, that was my first code in rust ever ;)

I actually never had the need to stop a running scripts, mine are quite short. So for me, a button is enough. Yes, HA allows to stop a running script, there is a `turn_on` & `turn_off` and also a `toggle` service. And what I did here, is just calling it by name, which is what I did so far in HA.
My personal opinion, I think its confusing to have this as a Switch, therefore I'd rather leave it as a Button. with only the the call by name.

**Review by zehnm — APPROVED** [review]
Thanks again! I'll merge it, button entitiy it is 👍 

You haven't picked the easiest Rust project to start with ;-)
Also some of my "earlier" work and a bit complicated with the Actor stuff.

**Comment by uvjim** [comment]
A quick question on this one now that it has been released... How do I pass parameters to the script that needs to be called?  I can't seem to find that in the Web Configurator.

**Comment by zehnm** [comment]
> How do I pass parameters to the script that needs to be called?

That is currently not possible.
For now, HA scripts can only be triggered. Supporting parameters will require more work, most likely with a new entity type, which supports dynamic parameters.

Since this PR is merged, let's continue in #20

**Pull request merged**